scripts/analysis_old.py

```diff
@@ -0,0 +1,1474 @@
+import collections
```

Comment:
Do not create this new file to hold the old file. The purpose of version control and code review is to always be able to look forward, while always having older versions available (via git, not via copy pasting into a new file).